### PR TITLE
Update QA to new inputs and datamodule

### LINKED ISSUE #955: Onboard old inputs to new `Input` object
By data type:

- [ ] audio ( #960 ): #1014 
- [x] graph: #953 
- [ ] image ( #961 ): #1015
- [x] pointcloud ( #962 ): #959 
- [x] tabular ( #963 )
- [ ] text ( #964 )
- [x] video: #953 

And integrations:

- [x] label studio ( #965 )

### LINKED ISSUE #964: Onboard text inputs to new Input object
Tracking issue #955

### PR DESCRIPTION
**Pull request by ethanwharris**
## What does this PR do?

Fixes #964 
Fixes #955

## Before submitting
- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/lightning-flash/tree/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? [not needed for typos/docs]
- [x] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/lightning-flash/blob/master/CHANGELOG.md)?

## PR review
 - [x] Is this pull request ready for review? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: flash_examples/question_answering.py [hunk 1]
```diff
@@ -21,13 +21,14 @@
...
     val_file="./data/squad_tiny/val.json",
+    batch_size=4,
 )
 
 # 2. Build the task
 model = QuestionAnsweringTask()
 
 # 3. Create the trainer and finetune the model
-trainer = Trainer(max_epochs=3, limit_train_batches=1, limit_val_batches=1)
+trainer = Trainer(max_epochs=3, fast_dev_run=True)
```
Comment (ethanwharris) [inline flash_examples/question_answering.py:31]:
```suggestion
trainer = Trainer(max_epochs=3)
```

File: flash/text/question_answering/data.py [hunk 2]
```diff
@@ -11,834 +11,270 @@
...
-            input_transform=input_transform,
-            val_split=val_split,
-            batch_size=batch_size,
-            num_workers=num_workers,
-            sampler=sampler,
-            **input_transform_kwargs,
+        return cls(
+            input_cls(RunningStage.TRAINING, train_file, transform=train_transform, **ds_kw),
+            input_cls(RunningStage.VALIDATING, val_file, transform=val_transform, **ds_kw),
+            input_cls(RunningStage.TESTING, test_file, transform=test_transform, **ds_kw),
```
Comment (tchaton) [inline flash/text/question_answering/data.py:244]: You are missing the predict input ?

File: flash/text/question_answering/input.py [hunk 3]
```diff
@@ -0,0 +1,362 @@
...
+                            """Dataset must contain either \"answer\" key as dict type or "answer_text" and
+                            "answer_start" as string and integer types."""
+                        )
+            if not isinstance(hf_dataset[self.answer_column_name][0], Dict):
+                raise TypeError(
+                    f'{self.answer_column_name} column should be of type dict with keys "text" and "answer_start"'
+                )
+
+        if flash._IS_TESTING:
+            # NOTE: must subset in this way to return a Dataset
```
Comment (tchaton) [inline flash/text/question_answering/input.py:225]: Dope !

**Review by tchaton — APPROVED** [review]
Looks dope !

File: flash/text/question_answering/data.py (hunk 2 above)
Comment (ethanwharris) [reply flash/text/question_answering/data.py:249]: Fixed :smiley:
